src/cats/labs/channel.cljc

```diff
@@ -78,64 +64,47 @@
 
     p/Functor
     (-fmap [_ f mv]
-      (let [c (a/chan 1 (map f))]
-        (a/pipe mv c)
-        c))
+      (a/pipe mv (a/chan 1 (map f))))
 
     p/Semigroup
     (-mappend [_ sv sv']
-      (chain-chans sv sv'))
+      (a/merge [sv sv']))
```

Comment:
Hmm, thank you very much for the clear explanation. I think that your first approach is the desired so no change is needed.